com.unity.render-pipelines.core/Runtime/Common/ConstantBuffer.cs

```diff
@@ -122,86 +122,134 @@ public static void ReleaseAll()
             m_RegisteredConstantBuffers.Clear();
         }
 
-        internal abstract class ConstantBufferBase
-        {
-            public abstract void Release();
-        }
-
         internal static void Register(ConstantBufferBase cb)
         {
             m_RegisteredConstantBuffers.Add(cb);
         }
 
-        class TypedConstantBuffer<CBType> : ConstantBufferBase where CBType : struct
-        {
-            // Used to track all global bindings used by this CB type.
-            HashSet<int> m_GlobalBindings = new HashSet<int>();
-            // Array is required by the ComputeBuffer SetData API
-            CBType[] m_Data = new CBType[1];
+    }
 
-            static TypedConstantBuffer<CBType> s_Instance = null;
-            internal static TypedConstantBuffer<CBType> instance
-            {
-                get
-                {
-                    if (s_Instance == null)
-                        s_Instance = new TypedConstantBuffer<CBType>();
-                    return s_Instance;
-                }
-                set
-                {
-                    s_Instance = value;
-                }
-            }
-            ComputeBuffer m_GPUConstantBuffer = null;
+    /// <summary>
+    /// The base class of Constant Buffer.
+    /// </summary>
+    public abstract class ConstantBufferBase
+    {
+        /// <summary>
+        /// Release the constant buffer.
+        /// </summary>
+        public abstract void Release();
+    }
 
-            TypedConstantBuffer()
-            {
-                m_GPUConstantBuffer = new ComputeBuffer(1, UnsafeUtility.SizeOf<CBType>(), ComputeBufferType.Constant);
-                ConstantBuffer.Register(this);
-            }
 
-            public void UpdateData(CommandBuffer cmd, in CBType data)
-            {
-                m_Data[0] = data;
+    /// <summary>
+    /// An instance of a constant buffer.
+    /// </summary>
+    /// <typeparam name="CBType">The type of structure representing the constant buffer data.</typeparam>
+    public class ConstantBuffer<CBType> : ConstantBufferBase where CBType : struct
+    {
+        // Used to track all global bindings used by this CB type.
+        HashSet<int> m_GlobalBindings = new HashSet<int>();
+        // Array is required by the ComputeBuffer SetData API
+        CBType[] m_Data = new CBType[1];
+
+
+        ComputeBuffer m_GPUConstantBuffer = null;
+
+        /// <summary>
+        /// Constant Buffer constructor.
+        /// </summary>
+        public ConstantBuffer()
+        {
+            m_GPUConstantBuffer = new ComputeBuffer(1, UnsafeUtility.SizeOf<CBType>(), ComputeBufferType.Constant);
+            ConstantBuffer.Register(this);
```

Comment:
Register() should only be called for Singleton constant buffers (as it's necessary for ReleaseAll to work).
Explicitly allocated CBs must be released explicitly as well.